core/src/main/java/org/apache/iceberg/Partition.java

```diff
@@ -0,0 +1,355 @@
+        return;
+      case 7:
+        this.eqDeleteRecordCount = (long) v;
+        return;
+      case 8:
+        this.eqDeleteFileCount = (int) v;
+        return;
+      case 9:
+        this.totalRecordCount = (long) v;
+        return;
+      case 10:
+        this.lastUpdatedAt = (long) v;
+        return;
+      case 11:
+        this.lastUpdatedSnapshotId = (long) v;
+        return;
+      default:
+        throw new UnsupportedOperationException("Unknown field ordinal: " + i);
+    }
+  }
+
+  @Override
+  public Object get(int i) {
+    switch (i) {
+      case 0:
+        return partitionData;
+      case 1:
+        return specId;
+      case 2:
+        return dataRecordCount;
+      case 3:
+        return dataFileCount;
+      case 4:
+        return dataFileSizeInBytes;
+      case 5:
+        return posDeleteRecordCount;
+      case 6:
+        return posDeleteFileCount;
+      case 7:
+        return eqDeleteRecordCount;
+      case 8:
+        return eqDeleteFileCount;
+      case 9:
+        return totalRecordCount;
+      case 10:
+        return lastUpdatedAt;
+      case 11:
+        return lastUpdatedSnapshotId;
+      default:
+        throw new UnsupportedOperationException("Unknown field ordinal: " + i);
+    }
+  }
+
+  @Override
+  public Schema getSchema() {
+    return prepareAvroSchema(partitionData.getPartitionType());
+  }
+
+  @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    } else if (!(o instanceof Partition)) {
+      return false;
+    }
+
+    Partition that = (Partition) o;
+    return partitionData.equals(that.partitionData)
+        && specId == that.specId
+        && dataRecordCount == that.dataRecordCount
+        && dataFileCount == that.dataFileCount
+        && dataFileSizeInBytes == that.dataFileSizeInBytes
+        && posDeleteRecordCount == that.posDeleteRecordCount
+        && posDeleteFileCount == that.posDeleteFileCount
+        && eqDeleteRecordCount == that.eqDeleteRecordCount
+        && eqDeleteFileCount == that.eqDeleteFileCount
+        && totalRecordCount == that.totalRecordCount
+        && lastUpdatedAt == that.lastUpdatedAt
+        && lastUpdatedSnapshotId == that.lastUpdatedSnapshotId;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        partitionData,
+        specId,
+        dataRecordCount,
+        dataFileCount,
+        dataFileSizeInBytes,
+        posDeleteRecordCount,
+        posDeleteFileCount,
+        eqDeleteRecordCount,
+        eqDeleteFileCount,
+        totalRecordCount,
+        lastUpdatedAt,
+        lastUpdatedSnapshotId);
+  }
+
+  public static org.apache.iceberg.Schema icebergSchema(Types.StructType partitionType) {
+    if (partitionType.fields().isEmpty()) {
+      throw new IllegalArgumentException("getting schema for an unpartitioned table");
+    }
+
+    return new org.apache.iceberg.Schema(
```

Comment:
Fields are based on the spec.
https://github.com/apache/iceberg/blob/main/format/spec.md#partition-statistics-file